src/clj/ataru/valinta_tulos_service/valintatulosservice_client.clj

```diff
@@ -93,13 +94,17 @@
   [cas-client ehto hakukohde-oid application-key if-unmodified-since]
   (let [url (url-helper/resolve-url
               :valinta-tulos-service.hyvaksynnan-ehto.hakukohteessa.hakemus hakukohde-oid application-key)
-        {:keys [status body]} (cas-client/cas-authenticated-put
+        response (cas-client/cas-authenticated-put
                                 cas-client url ehto (fn []
                                                       {:headers (if (some? if-unmodified-since)
                                                                   {"If-Unmodified-Since" if-unmodified-since}
-                                                                  {"If-None-Match" "*"}) }))]
-    (if (= 200 status)
-      (json/parse-string body true)
+                                                                  {"If-None-Match" "*"}) }))
+        {:keys [status body]} response
+        last-modified (get-header response "last-modified")]
+    (if (or (= 200 status) (= 201 status))
+      {:status status
+       :body (json/parse-string body true)
+       :headers {"Last-Modified" last-modified}}
       (throw (new RuntimeException (str "Could not put " url ", "
```

Comment:
handlerissa (:hyvaksynnan-ehto/set-ehto-hakukohteessa) on käsitelty myös statukset 400 404 ja 410 eli oletettu että niitä tulee läpi , pitäiskö nekin huomioida tässä?

Reply:
ehkäpä, mut nyt varmaan tärkeempää vaan saada korjaus eteenpäin. Mun mielestä (jos nyt ymmärsin oikein) kumpaakaan ei kyllä pitäis tulla jos kaikki toimii niinkun pitääkin, varmaan tarvis kaks käyttäjää samaan aikaan käsittelemään samoja tietoja niin et toinen poistaa ehdon ja toinen sen jälkeen yrittää muuttaa sitä